DS and Algorithm/DFS.py

```diff
@@ -21,17 +20,19 @@ def peek(self):
 
     def size(self):
         return len(self.items)
-    
+
 def dfs(graph, start):
     visited = set()
     stack = Stack()
-    
     stack.push(start)
+    
     while not stack.is_empty():
         vertex = stack.pop()
+        
         if vertex not in visited:
             visited.add(vertex)
             print(vertex, end=" ")  # Print visited vertex
+            
             # Add all neighbors to the stack
             for neighbor in graph[vertex]:
                 if neighbor not in visited:
```

Comment:
**Optimization Issue**: The dfs function's management of the visited set and stack does not efficiently check for duplicates before adding neighbors to the stack, potentially leading to unnecessary operations. <br> **Fix**: Modify the loop that adds neighbors to the stack to first check if they are in the visited set, reducing the number of unnecessary push operations to the stack. <br> **Code Suggestion**: 
 ```
 + if vertex not in visited:
 +     visited.add(vertex)
 +     print(vertex, end=" ")  # Print visited vertex
 +     for neighbor in graph[vertex]:
 +         if neighbor not in visited:
 +             stack.push(neighbor)
 ```